pkg/flowaggregator/s3uploader/s3uploader_test.go

```diff
@@ -180,30 +217,27 @@ func TestFlowRecordPeriodicCommit(t *testing.T) {
 		nameRand:         nameRand,
 		clusterUUID:      fakeClusterUUID,
 	}
-	testRecord := flowrecordtesting.PrepareTestFlowRecord()
-	s3UploadProc.writeRecordToBuffer(testRecord)
-	s3UploadProc.cachedRecordCount = 1
+	mockRecord := ipfixentitiestesting.NewMockRecord(ctrl)
+	flowaggregatortesting.PrepareMockIpfixRecord(mockRecord, true)
+	s3UploadProc.CacheRecord(mockRecord)
+	assert.Equal(t, 1, s3UploadProc.cachedRecordCount)
+
 	s3UploadProc.startExportProcess()
-	// wait for ticker to tick
-	err := wait.PollImmediate(10*time.Millisecond, 1*time.Second, func() (bool, error) {
-		mockS3Uploader.testReaderMutex.Lock()
-		defer mockS3Uploader.testReaderMutex.Unlock()
-		if mockS3Uploader.testReader.Len() != 0 {
-			return true, nil
-		}
-		return false, nil
-	})
-	assert.NoError(t, err)
+	assert.Eventually(t, func() bool {
+		wg.Wait()
+		return true
+	}, 200*time.Millisecond, 10*time.Millisecond)
```

Comment:
I gave bad advice here, because `wg.Wait` is blocking

instead you should use a channel.
```go
waitCh := make(chan struct{})
// ...
DoAndReturn(
    close(waitCh)
)
// ...
assert.Eventually(t, func() bool {
    select {
    case <- waitCh:
        // mock has been called
        return true
    default:
        // mock has not been called yet
        return false
    }
})
```
You may also want to increase the timeout to 500ms or even 1s, to be very conservative. The Windows test VMs have proved to be incredibly slow sometimes... It also doesn't impact the runtime of the tests because as long as the test is successful, it will complete very fast (except maybe on the Windows test VMs...).

Reply:
I can change to channel. But wg.Wait will not block forever if the timeout is reached. Is that what you are concerning about?

For example, I can get the following error:
```
Error Trace:	/antrea/pkg/flowaggregator/s3uploader/s3uploader_test.go:226
Error:      	Condition never satisfied
Test:       	TestFlowRecordPeriodicCommit
```

Reply:
It won't wait forever, but it's a very strange way to use `assert.Eventually`: the condition function will be called exactly once, and the tick period becomes meaningless